### PR TITLE
Add CI linting for some code style issues

### PR DESCRIPTION
**Pull request by speth**
Automate checks for some formatting issues that are easy to miss during pull request reviews.

**Changes proposed in this pull request**

- Check modified files for misleading Doxygen comment blocks that do not support "autobrief"
- Check for whitespace issues using `git diff --check`: trailing whitespace, tabs in indentation, and CRLF line endings.

**If applicable, provide an example illustrating new features this pull request is introducing**

CI will indicate failure and logs will include messages like the following for whitespace issues:
```
include/cantera/kinetics/Kinetics.h:48: trailing whitespace.
+//! reversible with a reverse rate satisfying detailed balance, include  
include/cantera/kinetics/Kinetics.h:55: trailing whitespace.
+//! A kinetics manager implements a kinetics model. Since the model equations  
include/cantera/kinetics/Kinetics.h:56: trailing whitespace.
+//! may be complex and expensive to evaluate, a kinetics manager may adopt 
```

And like this for bare `/*!` comment blocks (without a preceding `//!` block to provide the "brief" description):
```
------ include/cantera/kinetics/Kinetics.h ------


    /*!
     * Takes as input an array of properties for all species in the mechanism
     * and copies those values belonging to a particular phase to the output
     * array.
     * @param data Input data array.
     * @param phase Pointer to one of the phase objects participating in this
     *     reaction mechanism
     * @param phase_data Output array where the values for the the specified
     *     phase are to be written.
     * @deprecated Unused. To be removed after %Cantera 3.0.
     */
```

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by ischoegl** [comment]
#1561 is updated to preempt issues.

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          for file in ${{ steps.changed-cxx-files.outputs.all_changed_files }}; do
+            RESULT=`perl -0777 -ne 'while(m/\n\n *\/\*!.*?\*\//gs){print "$&\n";}' $file`
+            if [ "$RESULT" ]; then
+              echo ------ ${file} ------
+              echo "$RESULT"
+              RETCODE=1
+            fi
+          done
+          exit $RETCODE
+      - name: Whitespace errors
```
Comment (ischoegl) [inline .github/workflows/linters.yml:47]: If I read this correctly, this only applies to `*.cpp` and `*.h` files, although whitespace issues exist for other file types as well (as an aside, we also would like to see newlines at the end of files). Could you update the linting step so Python, MATLAB, etc. files are included in this check?

**Review by ischoegl — CHANGES_REQUESTED** [review]

File: .github/workflows/linters.yml (hunk 1 above)
Comment (speth) [reply .github/workflows/linters.yml:47]: The list of `.cpp` and `.h` files is only used for the Doxygen comment check. The whitespace check is just running on the diff between the PR base and head, and considers any file that Git thinks is a text file.

**Review by ischoegl — APPROVED** [review]
Thanks for the clarification - so, this looks great from my perspective!

**Pull request merged**